### PR TITLE
Remove python 2 version (if) checks from code

### PR DESCRIPTION
**Pull request by StephenSmith25**
**Description of work.**
This PR removes python 2 version checks from the codebase.

**To test:**
- Check builds and tests pass
- Check I haven't been overzealous with my removals

Contributes to #27722 

*This does not require release notes it is an internal change.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/PythonInterface/mantid/api/_workspaceops.py [hunk 1]
```diff
@@ -55,10 +55,6 @@ def op_wrapper(self, other):
     }
     # The division operator changed in Python 3
```
Comment (gui-co) [inline Framework/PythonInterface/mantid/api/_workspaceops.py:56]: This comment could be removed also.

File: scripts/Inelastic/IndirectImport.py [hunk 2]
```diff
@@ -117,12 +116,7 @@ def in_syspath(directory):
     # information and Python can import this without issue but it will take an untagged
     # filename in preference so we cannot keep the Python2/Python3 ones in the same
     # directory. We manipulate the sys.path temporarily to get the correct library.
```
Comment (gui-co) [inline scripts/Inelastic/IndirectImport.py:118]: Idem, I think this comment is useless now.

**Review by gui-co — COMMENTED** [review]
The build is fine, tests are passing. Maybe few changes in the comments.

I found other checks in the code that I think could be removed. I don't know if they are in the scope of this PR:
* `docs/source/algorithms/ExportSampleLogsToHDF5-v1.rst:37`
* `qt/python/mantidqt/utils/qt/testing/gui_window_test.py:24`
* `qt/python/mantidqt/widgets/test/test_fitpropertybrowserbase.py:21`
* `qt/python/mantidqt/widgets/codeeditor/test/test_execution.py:133`

**Comment by StephenSmith25** [comment]
Thanks for spotting those extra ones - and the comments! That should hopefully be everything done now.

**Review by gui-co — APPROVED** [review]
Thanks for the changes. Everything looks fine.

**Comment by martyngigg** [comment]
Great to get rid of these things!

**Pull request merged**